### PR TITLE
Rocm 3.7 rocm smi

### PR DESCRIPTION
**Pull request by gkathirv**

### CODE REVIEW

File: var/spack/repos/builtin/packages/miopen-hip/package.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    url = "https://github.com/ROCmSoftwarePlatform/MIOpen/archive/rocm-3.5.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+
+    version('3.5.0', sha256='aa362e69c4dce7f5751f0ee04c745735ea5454c8101050e9b92cc60fa3c0fb82')
+
+    variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
+
+    depends_on('cmake@3:', type='build')
+    depends_on('boost@1.58.0', type='link')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/miopen-hip/package.py]: This is the only version of boost that works?

File: var/spack/repos/builtin/packages/rocm-smi-lib/package.py [hunk 2]
```diff
@@ -13,17 +13,19 @@ class RocmSmiLib(CMakePackage):
...
 
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
+
     depends_on('cmake@3:', type='build')
 
     @run_after('install')
     def post_install(self):
         shutil.rmtree(self.prefix.lib)
-        install_tree(self.prefix.rocm_smi,  self.prefix)
+        copy_tree(self.prefix.rocm_smi,  self.prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocm-smi-lib/package.py]: Why the change here? The only difference between `copy_tree` and `install_tree` is how permissions are handled.

File: var/spack/repos/builtin/packages/rocm-smi/package.py [hunk 3]
```diff
@@ -5,30 +5,20 @@
...
+    version('3.5.0', sha256='4f46e947c415a4ac12b9f6989f15a42afe32551706b4f48476fba3abf92e8e7c')
 
-        popen('mkdir -p {0}/smi-test/tests'.format(prefix))
-        popen('cp -R {0}/tests/ {1}/smi-test/'.format(self.build_directory,
-                                                      prefix))
-        popen('cp -R {0}/test-rocm-smi.sh {1}/smi-test'.format(
-              self.build_directory, prefix))
+    def install(self, spec, prefix):
+        copy('rocm_smi.py', prefix)
+        symlink('rocm_smi.py', join_path(prefix, 'rocm_smi'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rocm-smi/package.py]:
```suggestion
        symlink('rocm_smi.py', prefix.rocm_smi)
```

**Review by adamjstewart — COMMENTED** [review]
@srekolam @arjun-raj-kuppala @haampie can you review?

File: var/spack/repos/builtin/packages/rocm-smi/package.py [hunk 4]
```diff
@@ -5,30 +5,20 @@
...
-    variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
 
-    phases = ['edit', 'build']
-
-    @run_after('build')
-    def post_build(self):
-        popen('cp -R {0}/rocm_smi.py {1}'.format(self.build_directory, prefix))
-        popen('ln -srf {0}/rocm_smi.py {1}/rocm-smi'.format(prefix, prefix))
+    version('3.7.0', sha256='4e34b3b4e409bb89677882f47d9988d56bc2d9bb9893f0712c22a4b73789e06a')
+    version('3.5.0', sha256='4f46e947c415a4ac12b9f6989f15a42afe32551706b4f48476fba3abf92e8e7c')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocm-smi/package.py:21]: You're missing a dependency on Python here:

```suggestion
    version('3.5.0', sha256='4f46e947c415a4ac12b9f6989f15a42afe32551706b4f48476fba3abf92e8e7c') 

    depends_on('python@3:', type='run')
```

File: var/spack/repos/builtin/packages/rocm-smi/package.py [hunk 5]
```diff
@@ -5,30 +5,20 @@
...
 
-        popen('mkdir -p {0}/smi-test/tests'.format(prefix))
-        popen('cp -R {0}/tests/ {1}/smi-test/'.format(self.build_directory,
-                                                      prefix))
-        popen('cp -R {0}/test-rocm-smi.sh {1}/smi-test'.format(
-              self.build_directory, prefix))
+    def install(self, spec, prefix):
+        copy('rocm_smi.py', prefix)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocm-smi/package.py:32]:
```suggestion
        filter_file(
            '^#!/usr/bin/python',
            '#!/usr/bin/env {0}'.format(
                os.path.basename(self.spec['python'].command.path)),
            'rocm_smi.py'
        )
        copy('rocm_smi.py', prefix)
```
This removes the system dependency on Python.

**Review by haampie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/miopen-hip/package.py (hunk 1 above)
Comment (haampie) [inline var/spack/repos/builtin/packages/miopen-hip/package.py]: The readme says at least 1.58, so this should work:

```suggestion
    depends_on('boost@1.58:', type='link')
```

File: var/spack/repos/builtin/packages/miopen-opencl/package.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    url = "https://github.com/ROCmSoftwarePlatform/MIOpen/archive/rocm-3.5.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala']
+
+    version('3.5.0', sha256='aa362e69c4dce7f5751f0ee04c745735ea5454c8101050e9b92cc60fa3c0fb82')
+
+    variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
+
+    depends_on('cmake@3:', type='build')
+    depends_on('boost@1.58.0', type='link')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/miopen-opencl/package.py]:
```suggestion
    depends_on('boost@1.58:', type='link')
```

**Review by haampie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/rocm-smi-lib/package.py (hunk 2 above)
Comment (gkathirv) [reply var/spack/repos/builtin/packages/rocm-smi-lib/package.py]: there was sync issue between my branch. reverted back to install_tree.

**Comment by gkathirv** [comment]
@adamjstewart Can you please, review the changes for rocm-smi?

**Review by adamjstewart — APPROVED** [review]
